packages/semcom-node/lib/store/services/pod-sync.service.spec.ts

```diff
@@ -0,0 +1,348 @@
+
+          if (url !== localPodUri) {
+
+            const pod = otherPods.find((podMock) => podMock.uri === url);
+            components = pod.components.map((component) => component.filename);
+            httpStatusCode = pod.httpStatus;
+
+          }
+
+          const quads: Quad[] = components.map((component) =>
+            createQuad(namedNode(component),
+              namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
+              namedNode('http://www.w3.org/ns/ldp#Resource')));
+
+          const bodyStream = serialize.serialize(streamify(quads), { contentType: 'text/turtle' });
+          const body = await streamToBuffer(bodyStream);
+
+          return new Response(body, { status: httpStatusCode, headers: { 'Content-Type': 'text/turtle' } });
+
+        } else {
+
+          // PodSyncService is requesting the metadata of a pod
+
+          const metadata = otherPods
+            .map((podMock) => podMock.components.find((component) => url === `${podMock.uri}${component.filename}`))
+            .find((mock) => mock !== undefined);
+
+          if (metadata) {
+
+            return new Response(
+              metadata.contents,
+              { status: metadata.httpStatus, headers: { 'Content-Type': 'text/turtle' } }
+            );
+
+          } else {
+
+            throw Error('pod is offline or non existent');
+
+          }
+
+        }
+
+      } else if (options.method === 'post' && url === localPodUri
+         && options.headers?.['Content-Type'] === 'text/turtle' && options.headers['Slug']) {
+
+        // already tests if headers are correct
+
+        localComponents.add(options.headers['Slug']);
+
+        return new Response('', { status: 201 });
+
+      } else {
+
+        return new Response('', { status: 501 });
+
+      }
+
+    });
+
+  };
+
+  beforeEach(() => {
+
+    localComponents = new Set();
+    jest.resetAllMocks();
+
+    store = new MemoryStore([ [ 'storage', [] ] ]);
+    podSyncService = new PodSyncService('storage', store, localPodUri);
+
+  });
+
+  it('should be correctly instantiated', () => {
+
+    expect(podSyncService).toBeTruthy();
+
+  });
+
+  describe('canHandle()', () => {
+
+    it('should be true', async () => {
+
+      await expect(podSyncService.canHandle().toPromise()).resolves.toBe(true);
+
+    });
+
+  });
+
+  describe('handle()', () => {
+
+    it('should always fetch its own pod', async () => {
+
+      mockPods([]);
+
+      await podSyncService.handle().toPromise();
+
+      expect(fetchMock).toBeCalledTimes(1);
+      expect(fetchMock.mock.calls[0][0]).toEqual(localPodUri);
+
+    });
+
+    it('fetches other pods', async () => {
+
+      const otherPods = [
+        {
+          uri: 'test.com/',
+          components: [],
+          isOnline: true,
+          httpStatus: 200,
+        },
+        {
+          uri: 'test2.com/',
+          components: [],
+          isOnline: true,
+          httpStatus: 200,
+        },
```

Comment:
set individual test pods as constants or declare them with let and define them in before each if you need to change properties.

Reply:
I won't add this comment to every test here, it's becoming a bit redundant :)
Change it where you see it.